libguard/guard_interface.cpp

```diff
@@ -167,9 +171,20 @@ GuardRecord create(const EntityPath& entityPath, uint32_t eId, uint8_t eType)
             }
             return getHostEndiannessRecord(existGuard);
         }
+
+        // Storing the 1st resolved guard record position
+        if (existGuard.recordId == GUARD_RESOLVED)
+        {
+            if (!isResolvedRecordPresent)
+            {
+                empPos = pos;
+                isResolvedRecordPresent = true;
+            }
+        }
+
+        id = be32toh(existGuard.recordId);
```

Comment:
Why you are preferring a new variable for this instead of using like previous code? Looks `id` is used in the `if` block.

Reply:
Yes, its required, as i was seeing issues with creating new recordId.

Reply:
It will be great if you mention the issue details in the comment section while saying some issues so that reviewer can understand. - Nevermind :)